features/environment.py

```diff
@@ -135,6 +140,26 @@ def __init__(
             if self.image_clean:
                 print(" Reuse_image specified, it will not be deleted.")
 
+        has_aws_keys = bool(aws_access_key_id and aws_secret_access_key)
+        if "pro" in self.machine_type:
+            self.aws_access_key_id = aws_access_key_id
```

Comment:
I can make this check more specific, but because of how we structure our CI with travis, the credentials environment variables are provided to all travis jobs, we only actually want to consume them if we are in the specific job that sets machine_type == 'pro.aws' or 'pro.azure'. Avoiding propagation of aws* credentials on non pro.aws machine_type allows us to skip in the `_should_skip_tags` check.